[Docs] Dropdown cpp code examples

mythrz opened the pull request:
- Dropdown cpp code examples
- Create Component  mentions how to convert a game component to ui component
- Working with UI components mentions how to edit cmakelists to set as build dependcy "Lyshine"

## Change summary

Revisiting the UI documentation for LyShine. Part of https://github.com/o3de/o3de.org/issues/2705
Noticed a few things:
- there was no mention on how to convert a "Game" gem to a "UI" gem
- there was no mention on how to edit the CMakeLists.txt to have LyShine as a build dependency
- there was no mention on how to include and inherit the bus
- there was no mention on how to override the functions/methods/events

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [Style Guide](https://o3de.org/docs/contributing/to-docs/style-guide/quick-reference)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

mythrz committed:
- Dropdown cpp code examples.
- Create Component  mentions how to convert a game component to ui component
- Working with UI components mentions how to edit cmakelists to set as build dependcy "Lyshine"

mythrz committed:
Update components-dropdown.md

typo on the *requestbus. 

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update components-dropdown.md

markdown with cpp so code is not only white on black

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update working.md

adding markdown cmake so the code is not only white on black

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

ShaunaGordon committed:
Merge branch 'development' into patch-3

ShaunaGordon commented:
@mythrz It looks like you missed signing a commit. Can you please go back and remedy that? Thanks!

mythrz committed:
Update components-dropdown.md

fixing relative path of link

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update working.md

fix for relative paths

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz committed:
Update working.md

DCO Remediation Commit for mythrz <44119535+mythrz@users.noreply.github.com>

I, mythrz <44119535+mythrz@users.noreply.github.com>, hereby add my Signed-off-by to this commit: b3b435a4285625bbbdd41037496d35ebaabc2b48

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

Signed-off-by: mythrz <44119535+mythrz@users.noreply.github.com>

mythrz commented:
Updated relative paths, and the bot has recognized the signed off/DCO of the latter.
Not sure how to edit the previous ones without redoing everything again

ShaunaGordon commented:
It now recognizes them all, so... 🤷🏼‍♀️
